types/types.cc

```diff
@@ -78,26 +78,19 @@ time_point_to_string(const T& tp)
         return fmt::format("{} milliseconds (out of range)", count);
     }
 
-    auto to_string = [] (const std::tm& tm) {
-        auto year_digits = tm.tm_year >= -1900 ? 4 : 5;
-        return fmt::format("{:-0{}d}-{:02d}-{:02d}T{:02d}:{:02d}:{:02d}",
-                tm.tm_year + 1900, year_digits, tm.tm_mon + 1, tm.tm_mday,
-                tm.tm_hour, tm.tm_min, tm.tm_sec);
-    };
-
     auto millis = d.rem;
-    if (!millis) {
-        return fmt::format("{}", to_string(tm));
-    }
     // adjust seconds for time points earlier than posix epoch
     // to keep the fractional millis positive
     if (millis < 0) {
         millis += 1000;
         seconds--;
         gmtime_r(&seconds, &tm);
     }
-    auto micros = millis * 1000;
-    return fmt::format("{}.{:06d}", to_string(tm), micros);
+
+    auto year_digits = tm.tm_year >= -1900 ? 4 : 5;
+    return fmt::format("{:-0{}d}-{:02d}-{:02d}T{:02d}:{:02d}:{:02d}.{:03d}Z",
```

Comment:
note to myself, we can drop the workaround in https://github.com/scylladb/scylla-dtest/blob/0324001e1d5275eb6d6a120018d51e7e0d4813e0/scylla_sstable_test.py#L176, once this change lands.

Reply:
In fact, the above test fails with this PR currently, so we will have to modify said test to work both with and without this PR, before this can go it. A CI is running currently and we should soon know the up-to-date list of dtest that sill require patching.

It is such a pain to do changes that require dtest changes too...

Reply:
yeah, i see. in the small time window. we need to at least appease both versions. lemme at least create a patch for the scylla_sstable_test.py . i really want to replace sstableloader in favor of `scylla dump-data`. as, IMO, a partial uuid-generation solution is not a solution ready for GA.

this should help us to partially address https://github.com/scylladb/scylla-tools-java/issues/333 -- one tiny step to ditch `tools/java`. lemme know if i can help.

Reply:
the dtest finished, see https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/3163/

- [ ] [scylla_sstable_test.TestScyllaSstableDumpataAllDatatypes.test_sstabledump_all_datatypes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/3163/testReport/junit/scylla_sstable_test/TestScyllaSstableDumpataAllDatatypes/FullDtest___full_split000___test_sstabledump_all_datatypes/) should be addressed by https://github.com/scylladb/scylla-dtest/pull/3328
- [ ] [update_cluster_layout_tests.TestUpdateClusterLayout.test_add_node_with_large_partition4](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/3163/testReport/junit/update_cluster_layout_tests/TestUpdateClusterLayout/FullDtest___full_split001___test_add_node_with_large_partition4/): i am still on it